uxarray/grid/intersections.py

```diff
@@ -115,3 +115,62 @@ def gca_gca_intersection(gca1_cart, gca2_cart, fma_disabled=False):
         res = np.append(res, x2)
 
     return res
+
+
+def gca_constLat_intersection(gca_cart,
+                              constLat,
+                              fma_disabled=False,
+                              verbose=False):
+    """Calculate the intersection point(s) of a Great Circle Arc (GCA) and a
+    constant latitude line in a Cartesian coordinate system.
+
+    To reduce relative errors, the Fused Multiply-Add (FMA) operation is utilized.
+    A warning is raised if the given coordinates are not in the cartesian coordinates, or
+    they cannot be accurately handled using floating-point arithmetic.
+
+    Parameters
+    ----------
+    gca_cart : [2, 3] np.ndarray Cartesian coordinates of the two end points GCA.
+    constLat : float
+        The constant latitude of the latitude line.
+    fma_disabled : bool, optional (default=False)
+        If True, the FMA operation is disabled. And a naive `np.cross` is used instead.
+
+    Returns
+    -------
+    np.ndarray
+        Cartesian coordinates of the intersection point(s).
+
+    Raises
+    ------
+    ValueError
+        If the input GCA is not in the cartesian [x, y, z] format.
+    """
+    constZ = np.sin(constLat)
+    x1, x2 = gca_cart
+    n = cross_fma(x1, x2)
```

Comment:
Can you add the equivalent if `fma_disabled` is True?

Reply:
To better achieve the performance, the `fma` should always be enabled as default